### ISSUE TITLE
app.isUnityRunning() returns true, although Unity is not even present

### ISSUE THREAD

**Issue by jayarjo** [electron/electron#18349]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

Docs say that [app.isUnityRunning()](https://electronjs.org/docs/api/app#appisunityrunning-linux) should tell whether the current desktop environment is Unity launcher. It return `true` in Ubuntu 18.04 which doesn't run Unity anymore.

This fact causes confusion around another method - [app.setBadgeCount()](https://electronjs.org/docs/api/app#appsetbadgecountcount-linux-macos) which is marked as supported in Linux wherever Unity is present.

**Comment by jnugh** [comment]
This is a workaround for https://github.com/electron/electron/issues/9046. Relevant implementation is here:
https://github.com/electron/electron/blob/6f5c850d60d05a188212f247e9edc746f6fce18c/lib/browser/init.ts#L183.
When I think about it, it would make a lot more sense API wise to query for features instead of environments. As a developer you should not have to care which environment you are running on.
Something like app.supportsBadgeCount() instead of isUnityRunning or similar.

**Comment by jayarjo** [comment]
The reason why I thought it was a bug, rather than a "feature" is because `setBadgeCount` relies on it underneath to decide whether it can show badge or not. `isUnityRunning` method tells it that it can and then it can't.

There's something broken and calling stuff wrong names only confuses it further. It reminds me of the situation wit User-Agent string, when browsers where faking their identity to make things "work".

**Comment by miniak** [comment]
Which versions of Electron does this reproduce with?

**Comment by jnugh** [comment]
Definitely master as chromium returns a Boolean for `isUnityRunning` based upon `XDG_CURRENT_DESKTOP` which is set to Unity if `currentPlatformSupportsAppIndicator` returns true which is the case for every Ubuntu desktop and `['Pantheon', 'Unity:Unity7', 'pop:GNOME']`

**Comment by codebytere** [comment]
I believe this is an upstream issue, in that at present `isUnityRunning` is implemented [here](https://github.com/electron/electron//blob/12b6a0f5b258142ab307c5d470586773ecf48355/atom/browser/browser_linux.cc#L141-L144) and maps straight onto [this](https://cs.chromium.org/chromium/src/chrome/browser/ui/libgtkui/unity_service.cc?sq=package:chromium&g=0&l=121) function in Chromium.

**Comment by jnugh** [comment]
You are right. I thought it uses GetDesktopEnvironment which itself reads the env variable I mentioned above. https://cs.chromium.org/chromium/src/base/nix/xdg_util.cc?q=Unity&sq=package:chromium&dr=C&l=57
So, simple test shows that libunity itself returns true when gnome is running
```c
#include <glib-object.h>
#include <unity/unity/unity.h>
#include <stdio.h>

int main(void) {
	printf("%i", unity_inspector_get_unity_running(unity_inspector_get_default())); // prints 1
}
```

There is an upstream report stating this also happens on KDE: https://bugs.launchpad.net/ubuntu/+source/libunity/+bug/1663747
And one for chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=594490 chromium switched to check `XDG_CURRENT_DESKTOP` in some cases.
